mathesar_ui/src/routes/schema-home/routes/datascape/table-view/header/header-cell/abstract-type-configuration/AbstractTypeConfiguration.svelte

```diff
@@ -38,38 +41,38 @@
     validationContext.validate();
   });
 
-  function resetAbstractType() {
-    selectedDbType = column.type;
-    typeOptions = { ...(column.type_options ?? {}) };
-    displayOptions = { ...(column.display_options ?? {}) };
-    defaultValue = column.default ? { ...column.default } : null;
-    selectedAbstractType = abstractTypeOfColumn;
+  function resetAbstractType(_column: Column) {
+    selectedAbstractType = abstractType;
+    selectedDbType = _column.type;
+    typeOptions = { ...(_column.type_options ?? {}) };
+    displayOptions = { ...(_column.display_options ?? {}) };
+    defaultValue = _column.default ? { ..._column.default } : null;
   }
-  resetAbstractType();
+  $: resetAbstractType(column);
 
   function clearTypeRelatedOptions() {
     typeOptions = {};
     displayOptions = {};
     defaultValue = null;
   }
 
-  function selectAbstractType(abstractType: AbstractType) {
-    if (selectedAbstractType !== abstractType) {
-      if (abstractType.identifier === abstractTypeOfColumn?.identifier) {
-        resetAbstractType();
-      } else if (abstractType.defaultDbType) {
-        selectedDbType = abstractType.defaultDbType;
+  function selectAbstractType(_abstractType: AbstractType) {
+    if (selectedAbstractType !== _abstractType) {
+      if (_abstractType.identifier === _abstractType?.identifier) {
```

Comment:
@seancolsen This is a hard to catch bug which was introduced because the names used were similar. We will have to add some form of detailed testing for this entire flow, not necessarily E2E.

The condition here should have been:
```suggestion
      if (_abstractType.identifier === abstractType?.identifier) {
```

Reply:
Ahh I see what happened here. Thanks for fixing this!